File: mobile/src/main/java/info/papdt/express/helper/ui/fragment/settings/SettingsContributors.java
package info.papdt.express.helper.ui.fragment.settings;

import android.os.Bundle;

import info.papdt.express.helper.R;
import moe.shizuku.preference.Preference;

public class SettingsContributors extends AbsPrefFragment implements Preference.OnPreferenceClickListener {

	private Preference mPrefCoderFox, mPrefHearSilent, mPrefArchieMeng;

	@Override
	public void onCreatePreferences(Bundle bundle, String s) {
		addPreferencesFromResource(R.xml.settings_contributors);

		mPrefCoderFox = findPreference("coderfox");
		mPrefCoderFox.setOnPreferenceClickListener(this);

		mPrefHearSilent = findPreference("hearsilent");
		mPrefHearSilent.setOnPreferenceClickListener(this);

		mPrefArchieMeng = findPreference("archiemeng");
		mPrefArchieMeng.setOnPreferenceClickListener(this);
	}

	@Override
	public boolean onPreferenceClick(Preference pref) {
		if (pref == mPrefCoderFox) {
			openWebsite(getString(R.string.contributors_coderfox_weibo_url));
			return true;
		}
		if (pref == mPrefHearSilent) {
			openWebsite("https://github.com/hearsilent");
			return true;
		}
		if (pref == mPrefArchieMeng) {
			openWebsite("https://github.com/ArchieMeng");
			return true;
		}
		return false;
	}
}

File: mobile/src/main/kotlin/info/papdt/express/helper/ui/fragment/settings/SettingsContributors.kt
package info.papdt.express.helper.ui.fragment.settings

import android.os.Bundle

import info.papdt.express.helper.R
import moe.shizuku.preference.Preference

class SettingsContributors : AbsPrefFragment(), Preference.OnPreferenceClickListener {

	private val mPrefCoderFox: Preference by PreferenceProperty("coderfox")
	private val mPrefHearSilent: Preference by PreferenceProperty("hearslient")
	private val mPrefArchieMeng: Preference by PreferenceProperty("archiemeng")

	override fun onCreatePreferences(bundle: Bundle?, s: String?) {
		addPreferencesFromResource(R.xml.settings_contributors)

		mPrefCoderFox.onPreferenceClickListener = this
		mPrefHearSilent.onPreferenceClickListener = this
		mPrefArchieMeng.onPreferenceClickListener = this
	}

	override fun onPreferenceClick(pref: Preference): Boolean {
		return when (pref) {
			mPrefCoderFox -> {
				openWebsite(getString(R.string.contributors_coderfox_weibo_url))
				true
			}
			mPrefHearSilent -> {
				openWebsite("https://github.com/hearsilent")
				true
			}
			mPrefArchieMeng -> {
				openWebsite("https://github.com/ArchieMeng")
				true
			}
			else -> false
		}
	}
}
